front/src/modules/auth/sign-in-up/hooks/useSignInUp.tsx

```diff
@@ -50,13 +52,47 @@ export function useSignInUp() {
   const [signInUpStep, setSignInUpStep] = useState<SignInUpStep>(
     SignInUpStep.Init,
   );
-  const [signInUpMode, setSignInUpMode] = useState<SignInUpMode>(
-    isMatchingLocation(AppPath.SignIn)
+  const [signInUpMode, setSignInUpMode] = useState<SignInUpMode>(() => {
+    if (isMatchingLocation(AppPath.Invite)) {
+      return SignInUpMode.Invite;
+    }
+
+    return isMatchingLocation(AppPath.SignIn)
       ? SignInUpMode.SignIn
-      : SignInUpMode.SignUp,
-  );
+      : SignInUpMode.SignUp;
+  });
   const [showErrors, setShowErrors] = useState(false);
 
+  const [workspaceFromInviteHashQuery, { data: workspace }] =
+    useGetWorkspaceFromInviteHashLazyQuery();
+
+  useEffect(() => {
+    function navigateToSignUp() {
+      enqueueSnackBar('workspace does not exist', {
```

Comment:
we should move this logic to AuthAutoRouter (or to a hook that would be called in AuthAutoRouter).
We avoid adding useEffect as they cause re-render. We only allow one in AuthAutoRouter. Let's move this there, you'll see that there is already some redirect logic there

Reply:
Hello there, @charlesBochet!

Primarily, this handles retrieving workspace details. If an error crops up with the page, it leads to navigation towards the sign-up page, which is a secondary function.

it is also important that the `workspaceInviteHash` does not change during the lifetime of this component so we do not expect the hook to run more than once.

Shifting this logic to the AuthAutoRouter necessitates having a means to store the fetched workspace in a state, thereby enabling the useSignInUp hook to reach this data.

Reply:
why don't you fetch the workspace at the level of the router directly (based on inviteHash)? And in useSinInUp? As we are relying on apollo cache, this will not perform two queries!

Reply:
This makes sense